### ISSUE TITLE
rerun `yoshi start` on configuration change.

### ISSUE THREAD

**Issue by ranyitz** [wix/yoshi#1408]
# 🙋 Feature Request

### 😯 Current Behavior

When configuring Yoshi via `package.json`/`yoshi.config.js` during `yoshi start`, the user needs to manually press `ctrl+c` and then run `yoshi start` again so this new configuration will take effect.

### 🤔 Expected Behavior

When the developer changes a configuration that needs a rerun in order to take effect let's do automatically it for him.
 
### 💁 Possible Solution

During `yoshi start`, listen to changes on `package.json` `yoshi` field and `yoshi.config.js` files and rerun accordingly.

### 🔦 Context

I've seen many developers struggle when trying to change configuration options (like `ssl`) when they don't know that they need to close and rerun Yoshi.

We aim towards something you start once, and doesn't need to touch it again. This feature will take us one step closer.

**Comment by saarkuriel** [comment]
Can I try and take this issue? @ranyitz @ronami

**Comment by ranyitz** [comment]
@saarkuriel Yes of course! Let us know if you need any help :1st_place_medal:

**Comment by ranyitz** [comment]
@Gongreg Asked me if he can take this issue, and suggested to run Yoshi as a subprocess so it will be easy to kill and restart it.

This is a good solution and was my initial goto as well.
The problem with running start in a different process is that we need to wait around 100ms more just because we now want to maybe restart the process (which is not very common thing to do).

I asked @ronami and we thought about how to do it without starting a new process. I wrote some pseudo-code that show how it can be done.

### The changes that needs to be done:

* Dev environment needs to expose a `stop` method, this stop method should be able to stop both [`webpack-dev-server` & `app-server-process`](https://github.com/wix/yoshi/blob/master/packages/yoshi-common/src/dev-environment.ts#L41-L42) which I think already has a `stop`/`end`/`kill` method.

* `Start` command needs to expose a way to stop it, a good way to do it is to return a function which will be called when we want to stop the command.

![image](https://user-images.githubusercontent.com/11733036/72439432-995c7800-37af-11ea-99a8-d7640c062aaf.png)

* The `yoshi-app` file can listen to changes on the config and in case it has been changed, stop the current dev environment and create a new one.

![image](https://user-images.githubusercontent.com/11733036/72439116-f60b6300-37ae-11ea-8a17-df370972e77f.png)

* We can also send a message to the screen that we're currently reloading the `dev-environment` which would be really nice.

I haven't tested it, this is all just pseudo-code, and reality may show us some obstacles.

@Gongreg Let me know if you think it's a good idea and if you want to take this feature.

**Comment by ranyitz** [comment]
@Gongreg What do you say?

**Comment by Gongreg** [comment]
@ranyitz, I prefer to try your solution, it definitely sounds better than starting yoshi as a subproccess. We just need to see if it will work. I haven't had time to work on it though yet, sorry.

**Comment by ranyitz** [comment]
It's ok, I'm assigning you to this issue. 🙂

**Comment by Gongreg** [comment]
After talking yesterday with @ranyitz, we've decided to close this issue.

The complexity we get by introducing this feature far outweighs any benefit we get from it. There are only few config options that could be actually useful, but if we watched only few of those options this functionality will be not self evident. The best thing to do now is to learn from this case and make sure we avoid situation like this in the future.
